skyportal/models.py

```diff
@@ -923,9 +924,11 @@ def get_obj_if_owned_by(obj_id, user_or_token, options=[]):
             # view, raise AccessError
             raise
     if obj is None:  # There is no associated Source/Cand, so check based on photometry
-        if Obj.get_photometry_owned_by_user(obj_id, user_or_token):
-            return Obj.query.options(options).get(obj_id)
-        raise AccessError("Insufficient permissions.")
+        if Obj.query.get(obj_id).photometry:  # Obj has associated Photometry
+            if Obj.get_photometry_owned_by_user(obj_id, user_or_token):
```

Comment:
Why not have `get_photometry_owned_by_user` return empty if no photometry exists?

Reply:
@stefanv @dannygoldstein it does...

Reply:
that's exactly what's being checked here

Reply:
Yes, but why not fix it in the `Obj.get_photometry_owned_by_user` call?

Reply:
I'm not sure I follow. What do you mean fix it? This call returns all associated photometry a user has access to, or [] if that is no photometry rows.

Reply:
Thus this is checking if an empty list is returned or not